pallets/messages/src/mock.rs

```diff
@@ -147,12 +147,19 @@ impl AccountProvider for AccountHandler {
 		Ok(get_msa_from_account(*key))
 	}
 
-	fn ensure_valid_delegation(provider: Provider, _delegator: Delegator) -> DispatchResult {
+	fn ensure_valid_delegation(
+		provider: Provider,
+		_delegator: Delegator,
+	) -> Result<ProviderInfo<Self::BlockNumber, Self::MaxSchemaGrants>, DispatchError> {
 		if provider == Provider(2000) {
 			return Err(DispatchError::Other("some delegation error"))
 		};
 
```

Comment:
some unit tests much have expected error (negative tests), not sure how they passed or probably never existed

Reply:
We saw a lot of that. Most of that needs to be fixed in #452 as the (current) mocks are really making it hard to test nicely.

Reply:
A common test runtime for all pallets